Create separated npm package for CLI functionality

#### Perceived Problem

The 7.0.0 version introduced the CLI functionality that able to generate a new typed client. It is a nice feature, if you use it. I don't use it because most of the time I use only 1-2 query and/or mutation in the micro services and I don't need more hundred types and operations.
I started to use `grapqhl-request` because it provided the nice, solid core graphql functionality. I highly appreciate for this package and thank you for it.

`dprint`, `@dprint/*`, `@molt/command`, `zod` packages are required for this functionality. These dependencies add 20+ MB to the docker images. It increases the docker image size by 50-100% in my cases.

I don't use any bundler, because it just complicate the things. I would like to ask to move the CLI feature into a separated npm package and if someone would like to use it then they can add it to the dev dependencies.

#### Ideas / Proposed Solution(s)

Create separated npm package for cli feature.

Upgrading `graphql-request` v7 led to this error:
```
-    "graphql-request": "^6.1.0",
+    "graphql-request": "^7.0.1",
...
#0 205.5 npm ERR! command failed
#0 205.5 npm ERR! command sh -c node ./install.js
#0 205.5 npm ERR! node:internal/modules/cjs/loader:1227
#0 205.5 npm ERR!   throw err;
#0 205.5 npm ERR!   ^
#0 205.5 npm ERR! 
#0 205.5 npm ERR! Error: Cannot find module '@dprint/linux-x64-musl/package.json'
#0 205.5 npm ERR! Require stack:
```

--- a/README.md
+++ b/README.md
@@ -12,7 +12,6 @@ Minimal GraphQL client supporting Node and browsers for scripts or simple apps.
   - [TypeScript Setup](#typescript-setup)
 - [Quick Start](#quick-start)
 - [Examples](#examples)
-- [Usage Trend](#usage-trend)
 - [Node Version Support](#node-version-support)
 - [Reference](#reference)
   - [Configuration](#configuration)
@@ -26,6 +25,9 @@ Minimal GraphQL client supporting Node and browsers for scripts or simple apps.
   - [Why do I have to install `graphql`?](#why-do-i-have-to-install-graphql)
   - [Do I need to wrap my GraphQL documents inside the `gql` template exported by `graphql-request`?](#do-i-need-to-wrap-my-graphql-documents-inside-the-gql-template-exported-by-graphql-request)
   - [What sets `graphql-request` apart from other clients like Apollo, Relay, etc.?](#what-sets-graphql-request-apart-from-other-clients-like-apollo-relay-etc)
+- [Project Stats](#project-stats)
+  - [Package Installs](#package-installs)
+  - [Repo Beats](#repo-beats)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
@@ -120,12 +122,6 @@ await client.request(document)
   - [Middleware](./examples/other-middleware.ts)
   - [Error Handling](./examples/other-error-handling.ts)
 
-## Usage Trend
-
-[Usage Trend of graphql-request](https://npm-compare.com/graphql-request/#timeRange=THREE_YEARS)
-
-![image](https://github.com/jasonkuhrt/graphql-request/assets/3455798/8d27c215-f20f-46f9-b38d-61f41d14882f)
-
 ## Node Version Support
 
 We only (officially) support [versions of Nodejs](https://github.com/nodejs/Release#release-schedule) of the following status:
@@ -204,3 +200,15 @@ No. It is there for convenience so that you can get the tooling support like aut
 `graphql-request` is the most minimal and simplest to use GraphQL client. It's perfect for small scripts or simple apps.
 
 Compared to GraphQL clients like Apollo or Relay, `graphql-request` doesn't have a built-in cache and has no integrations for frontend frameworks. The goal is to keep the package and API as minimal as possible.
+
+## Project Stats
+
+### Package Installs
+
+<a href="https://npm-compare.com/graphql-request#timeRange=FIVE_YEARS" target="_blank">
+  <img src="https://npm-compare.com/img/npm-trend/FIVE_YEARS/graphql-request.png" width="100%" alt="NPM Usage Trend of graphql-request" />
+</a>
+
+### Repo Beats
+
+![Alt](https://repobeats.axiom.co/api/embed/aeb7beaee43b190e90868357c5a2898f517fb63e.svg "Repobeats analytics image")

--- a/README.md
+++ b/README.md
@@ -12,6 +12,7 @@ Minimal GraphQL client supporting Node and browsers for scripts or simple apps.
   - [TypeScript Setup](#typescript-setup)
 - [Quick Start](#quick-start)
 - [Examples](#examples)
+- [Usage Trend](#usage-trend)
 - [Node Version Support](#node-version-support)
 - [Reference](#reference)
   - [Configuration](#configuration)

--- a/package.json
+++ b/package.json
@@ -80,15 +80,26 @@
     "release:pr": "dripip pr"
   },
   "dependencies": {
-    "@dprint/formatter": "^0.3.0",
-    "@dprint/typescript": "^0.91.1",
     "@graphql-typed-document-node/core": "^3.2.0",
     "@molt/command": "^0.9.0",
-    "dprint": "^0.46.2",
     "zod": "^3.23.8"
   },
   "peerDependencies": {
-    "graphql": "14 - 16"
+    "graphql": "14 - 16",
+    "dprint": "^0.46.2",
+    "@dprint/formatter": "^0.3.0",
+    "@dprint/typescript": "^0.91.1"
+  },
+  "peerDependenciesMeta": {
+    "dprint": {
+      "optional": true
+    },
+    "@dprint/formatter": {
+      "optional": true
+    },
+    "@dprint/typescript": {
+      "optional": true
+    }
   },
   "devDependencies": {
     "@arethetypeswrong/cli": "^0.15.3",

--- a/src/layers/2_generator/files.ts
+++ b/src/layers/2_generator/files.ts
@@ -1,5 +1,4 @@
-import { createFromBuffer } from '@dprint/formatter'
-import { getPath } from '@dprint/typescript'
+import type { Formatter } from '@dprint/formatter'
 import _ from 'json-bigint'
 import fs from 'node:fs/promises'
 import * as Path from 'node:path'
@@ -18,6 +17,16 @@ export interface Input {
   errorTypeNamePattern?: OptionsInput['errorTypeNamePattern']
 }
 
+const getTypeScriptFormatter = async (): Promise<Formatter | undefined> => {
+  try {
+    const { createFromBuffer } = await import(`@dprint/formatter`)
+    const { getPath } = await import(`@dprint/typescript`)
+    return createFromBuffer(await fs.readFile(getPath()))
+  } catch (error) {
+    return undefined
+  }
+}
+
 export const generateFiles = async (input: Input) => {
   const sourceDirPath = input.sourceDirPath ?? process.cwd()
   const schemaPath = input.schemaPath ?? Path.join(sourceDirPath, `schema.graphql`)
@@ -31,7 +40,7 @@ export const generateFiles = async (input: Input) => {
     customScalarCodecsFilePath.replace(/\.ts$/, `.js`),
   )
   const customScalarCodecsPathExists = await fileExists(customScalarCodecsFilePath)
-  const typeScriptFormatter = (input.format ?? true) ? createFromBuffer(await fs.readFile(getPath())) : undefined
+  const typeScriptFormatter = (input.format ?? true) ? await getTypeScriptFormatter() : undefined
 
   const codes = generateCode({
     name: input.name,

